wapitiCore/attack/mod_network_device.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+from typing import Optional
+
+from bs4 import BeautifulSoup
+from httpx import RequestError
+
+from wapitiCore.attack.attack import Attack
+from wapitiCore.net import Request
+from wapitiCore.net.response import Response
+from wapitiCore.definitions.fingerprint import NAME as TECHNO_DETECTED, WSTG_CODE
+
+from wapitiCore.main.log import log_blue
+
+MSG_TECHNO_VERSIONED = "{0} {1} detected"
+MSG_NO_UBIKA = "No UBIKA Detected"
+
+
+class ModuleNetworkDevice(Attack):
+    """Base class for detecting version."""
+    name = "network_device"
+    version = ""
+
+    async def check_ubika(self, url):
+        check_list = ['app/monitor/']
+        for item in check_list:
+            request = Request(f'{url}{item}', 'GET')
+            try:
+                response: Response = await self.crawler.async_send(request, follow_redirects=True)
+            except RequestError:
+                self.network_errors += 1
+            else:
+                soup = BeautifulSoup(response.content, 'html.parser')
+                title_tag = soup.title
+                return response.is_success and title_tag and "UBIKA" in title_tag.text.strip()
+
+    async def get_ubika_version(self, url):
+        version_uri = "app/monitor/api/info/product"
+        request = Request(f'{url}{version_uri}', 'GET')
+        try:
+            response: Response = await self.crawler.async_send(request, follow_redirects=True)
+        except RequestError:
```

Comment:
Same here, your error is catched by the try/except and does not return anything in case of error.